lib/cocoapods/installer/target_installer/pod_target_installer.rb

```diff
@@ -66,9 +67,13 @@ def add_files_to_build_phases
           native_target.add_file_references(header_file_refs) do |build_file|
             # Set added headers as public if needed
             if target.requires_frameworks?
+              build_file.settings ||= {}
               if public_headers.include?(build_file.file_ref.real_path)
-                build_file.settings ||= {}
                 build_file.settings['ATTRIBUTES'] = ['Public']
+              elsif private_headers.include?(build_file.file_ref.real_path)
+                build_file.settings['ATTRIBUTES'] = ['Private']
+              else
+                build_file.settings['ATTRIBUTES'] = ['Project']
```

Comment:
>as all headers in `source_files` which aren't public are private now

They would be even more encapsulated by being project headers, so they aren't exposed at all.
Private headers are only those, which are explicitly declared as such.

So this PR would add the following rule:
* If there are any explicit `public_headers`, all headers in `source_files`, which are not declared as `private_headers`, will be project headers.